### PR TITLE
fix(sourcemaps): add react sourcemap plugin

### PR DESCRIPTION
**Pull request by JonasBa**
Adds react sourcemap plugin

### CODE REVIEW

**Comment by scttcper** [comment]
do the sourcemaps work with react 17?

File: yarn.lock [hunk 1]
```diff
@@ -11025,7 +11192,7 @@ word-wrap@~1.2.3:
   resolved "https://registry.yarnpkg.com/word-wrap/-/word-wrap-1.2.4.tgz#cb4b50ec9aca570abd1f52f33cd45b6c61739a9f"
   integrity sha512-2V81OA4ugVo5pRo46hAoD2ivUJx8jXmWXfUkY4KFNw0hEptvN0QfH3K4nHiwzGeKl5rFKedV48QVoqYavy4YpA==
 
-wrap-ansi@^7.0.0:
+"wrap-ansi-cjs@npm:wrap-ansi@^7.0.0", wrap-ansi@^7.0.0:
```
Comment (scttcper) [inline yarn.lock:11195]: this looks goofy

File: yarn.lock [hunk 2]
```diff
@@ -10107,7 +10243,7 @@ string-length@^4.0.1:
     char-regex "^1.0.2"
     strip-ansi "^6.0.0"
 
-"string-width@^1.0.2 || 2 || 3 || 4", string-width@^4.1.0, string-width@^4.2.0, string-width@^4.2.3:
+"string-width-cjs@npm:string-width@^4.2.0", "string-width@^1.0.2 || 2 || 3 || 4", string-width@^4.1.0, string-width@^4.2.0, string-width@^4.2.3:
```
Comment (scttcper) [inline yarn.lock:10246]: same here

File: yarn.lock [hunk 3]
```diff
@@ -1482,6 +1502,18 @@
...
+  version "8.0.2"
+  resolved "https://registry.yarnpkg.com/@isaacs/cliui/-/cliui-8.0.2.tgz#b37667b7bc181c168782259bab42474fbf52b550"
+  integrity sha512-O8jcjabXaleOG9DQ0+ARXWZBTfnP4WNAqzuiJK7ll44AmxGKv/J2M4TPjxjY3znBCfvBXFzucm1twdyFybFqEA==
+  dependencies:
+    string-width "^5.1.2"
+    string-width-cjs "npm:string-width@^4.2.0"
+    strip-ansi "^7.0.1"
+    strip-ansi-cjs "npm:strip-ansi@^6.0.1"
+    wrap-ansi "^8.1.0"
+    wrap-ansi-cjs "npm:wrap-ansi@^7.0.0"
```
Comment (scttcper) [inline yarn.lock:1515]: seems to be where they're coming from, haven't seen this version style used much

**Comment by JonasBa** [comment]
@scttcper yeah, sourcemaps will work with react 17, I just had to update the plugin build to make it esm/cjs compact.

**Comment by evanpurkhiser** [comment]
what does it do?

**Comment by JonasBa** [comment]
@evanpurkhiser it adds sources for react and react-dom (bc they ship without), you can see the changes in https://github.com/sentry-demos/empower/pull/287

**Review by scttcper — APPROVED** [review]

**Pull request merged**